proto/api/v1/api.proto

```diff
@@ -828,34 +828,42 @@ message GetPluginResponse {
 }
 
 message PluginSpecifications{
+  message Parameter {
+    message Validation{
+      enum Type {
+        TYPE_UNSPECIFIED = 0;
+        // parameter must be present
+        TYPE_REQUIRED = 1;
+        // parameter must be greater than {value}
+        TYPE_GREATER_THAN = 2;
+        // parameter must be less than {value}
+        TYPE_LESS_THAN = 3;
+        // parameter must be included in the comma separated list {value}
+        TYPE_INCLUSION = 4;
+        // parameter must not be included in the comma separated list {value}
+        TYPE_EXCLUSION = 5;
+        // parameter must match the regex {value}
+        TYPE_REGEX = 6;
+      }
 
-  message Validation{
-    enum Type {
-      TYPE_UNSPECIFIED = 0;
-      TYPE_REQUIRED = 1;
-      TYPE_INCLUSION = 2;
-      TYPE_EXCLUSION = 3;
-      TYPE_GREATER_THAN = 4;
-      TYPE_LESS_THAN = 5;
-      TYPE_REGEX = 6;
+      Type validation =1;
```

Comment:
Nitpick: I'd name this `type` so we have a `Validation.type` instead of `Validation.validation`.
```suggestion
      Type type = 1;
```